packages/image_picker/image_picker_ios/ios/Classes/FLTImagePickerImageUtil.m

```diff
@@ -28,16 +28,36 @@ - (instancetype)initWithImages:(NSArray<UIImage *> *)images interval:(NSTimeInte
 
 @implementation FLTImagePickerImageUtil : NSObject
 
+static UIImage *FLTImagePickerDrawScaledImage(UIImage *imageToScale, double width, double height) {
+  UIGraphicsImageRenderer *imageRenderer =
+      [[UIGraphicsImageRenderer alloc] initWithSize:CGSizeMake(width, height)
+                                             format:imageToScale.imageRendererFormat];
+  return [imageRenderer imageWithActions:^(UIGraphicsImageRendererContext *rendererContext) {
+    // Flip vertically to translate between UIKit and Quartz.
+    CGContextTranslateCTM(rendererContext.CGContext, 0, height);
+    CGContextScaleCTM(rendererContext.CGContext, 1, -1);
+    CGContextDrawImage(rendererContext.CGContext, CGRectMake(0, 0, width, height),
+                       imageToScale.CGImage);
+  }];
+}
+
 + (UIImage *)scaledImage:(UIImage *)image
                 maxWidth:(NSNumber *)maxWidth
                maxHeight:(NSNumber *)maxHeight
      isMetadataAvailable:(BOOL)isMetadataAvailable {
   double originalWidth = image.size.width;
   double originalHeight = image.size.height;
-  double aspectRatio = originalWidth / originalHeight;
 
-  bool hasMaxWidth = maxWidth != nil;
-  bool hasMaxHeight = maxHeight != nil;
+  BOOL hasMaxWidth = maxWidth != nil;
+  BOOL hasMaxHeight = maxHeight != nil;
+
+  if ((originalWidth == maxWidth.doubleValue && originalHeight == maxHeight.doubleValue) ||
+      (!hasMaxWidth && !hasMaxHeight)) {
+    // Nothing to scale.
+    return image;
+  }
```

Comment:
This code is new, and unrelated to the fix.  If there is nothing to scale, return the original image instead of doing extra work.